### ISSUE TITLE
Kernel/Devices/SB16: ASSERTION FAILED: length <= PAGE_SIZE: freezes system

### ISSUE THREAD

**Issue by bcoles** [SerenityOS/serenity#4557]
Reproduce:

```
python -c "print('A' * 4096)" > aaa
cat aaa > /dev/audio
# or:
cat /dev/random > /dev/audio
```

Log:

```
[cat(68:68)]: ASSERTION FAILED: length <= PAGE_SIZE
../Kernel/Devices/SB16.cpp:249 in virtual Kernel::KResultOr<long unsigned int> Kernel::SB16::write(Kernel::FileDescription&, size_t, const Kernel::UserOrKernelBuffer&, size_t)
[cat(68:68)]: 0xc0119076  __assertion_failed(char const*, char const*, unsigned int, char const*) +185
[cat(68:68)]: 0xc01299d7  Kernel::SB16::write(Kernel::FileDescription&, unsigned long, Kernel::UserOrKernelBuffer const&, unsigned long) +201
[cat(68:68)]: 0xc0138f5f  Kernel::FileDescription::write(Kernel::UserOrKernelBuffer const&, unsigned long) +149
[cat(68:68)]: 0xc019634e  Kernel::Process::do_write(Kernel::FileDescription&, Kernel::UserOrKernelBuffer const&, unsigned long) +380
[cat(68:68)]: 0xc0196828  Kernel::Process::sys$write(int, unsigned char const*, long) +292
[cat(68:68)]: 0xc017a110  syscall_handler +1724
[cat(68:68)]: 0xc0179a4c  syscall_asm_entry +49
```

**Comment by tomuta** [comment]
I believe I addressed this problem in #4339, but the PR isn't ready yet as the DMA locks up the system for some reason.

**Comment by bcoles** [comment]
Fixed in #11039.
